WalletWasabi/Extensions/RPCClientExtensions.cs

```diff
@@ -109,6 +109,7 @@ public static async Task<AllFeeEstimate> EstimateAllFeeAsync(this IRPCClient rpc
 			return rpcFeeEstimationTasks
 				.Where(x => x.IsCompletedSuccessfully)
 				.Select(x => x.Result)
+				.DistinctBy(x => x.Blocks)
```

Comment:
This solution assumes the `Blocks` come in order and are the correspond with the targets that we requested, what is not true. My suggestion is ignoring the `Blocks` as we did in the previous implementation and use something like this:

```c#
			return rpcFeeEstimationTasks
				.Zip(Constants.ConfirmationTargets, (task, target) => (task, target))
				.Where(x => x.task.IsCompletedSuccessfully)
				.Select(x => (target: x.target, feeRate: x.task.Result.FeeRate))
				.ToDictionary(x => x.target, x => (int)Math.Ceiling(x.feeRate.SatoshiPerByte));
```

Reply:
Shall we use ZipForceEqualLength?